src/mscorlib/shared/System/Collections/HashHelpers.cs

```diff
@@ -104,5 +104,21 @@ public static int ExpandPrime(int oldSize)
                 return s_serializationInfoTable;
             }
         }
+
+        internal static int[] ResizeDictionaryArrays(Array _entries, Array entries, int count, int newSize)
+        {
+            // Identical initialize code from Dictionary Resize, externalized to not replicate from generic instantiation
+            int[] buckets = new int[newSize];
+            for (int i = 0; i < buckets.Length; i++)
```

Comment:
I would omit this change from this PR. We have rejected similar micro-optimizations that make the code hard to follow and maintains.

If we were to reconsider them, I think switching buckets to have one-based index so that it does not need to be initialized to -1 would look better than this and likely save more.